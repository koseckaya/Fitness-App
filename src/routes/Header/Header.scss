@import '../../styles/vars.scss';
.header {
  box-sizing: border-box;
  background: $gray;
  padding: 15px 20px;
 
  &__container {
    display: flex;
    align-items: center;
    gap: 1%;
  }
  &__signUp{
    display: flex;
    gap: 10px;
    width: 18%;
    justify-content: flex-end;
    padding-right: 1rem;
  }
  .nav {
    display: flex;
    width: 55%;
    justify-content: space-around;
  }
}

.logo {
  width: 25%;
  display: flex;
  justify-content: center;
  align-items: center;
}

a {
  text-decoration: none;
  color: $color-primary;
  text-transform: capitalize;
}

.header__signUp .button {
  width: auto;
}

@include for-max-1250px() {
  .header {
    .nav {
      height: 60px;
        display: grid;
        justify-content: center;
        align-items: center;
        width: 55%;
        justify-content: space-evenly;
        row-gap: 1.2rem;
        grid-template-areas:
        'A A B B C C'
        'D D D E E E';
      a {
        justify-self: center;
        font-size: 1.2rem;
      }
      .nav-item.a {
        grid-area: A;
      }
      .nav-item.b {
        grid-area: B;
      }
      .nav-item.c {
        grid-area: C;
      }
      .nav-item.d {
        grid-area: D;
        justify-self: end;
        padding-right: 1rem;
      }
      .nav-item.e {
        grid-area: E;
        justify-self: start;
        padding-left: 1rem;
      }
    }
  }
}

.header__hamburger {
  position: absolute;
  z-index: 9999;
  right: 3rem;
  display: none !important;
  background-color: transparent;
  border: none;
  cursor: pointer;
  display: flex;
  padding: 0;
}
.line {
  fill: none;
  stroke: black;
  stroke-width: 6;
  transition: stroke-dasharray 600ms cubic-bezier(0.4, 0, 0.2, 1),
    stroke-dashoffset 600ms cubic-bezier(0.4, 0, 0.2, 1);
}
.line1 {
  stroke-dasharray: 60 207;
  stroke-width: 6;
}
.line2 {
  stroke-dasharray: 60 60;
  stroke-width: 6;
}
.line3 {
  stroke-dasharray: 60 207;
  stroke-width: 6;
}
.opened .line1 {
  stroke-dasharray: 90 207;
  stroke-dashoffset: -134;
  stroke-width: 6;
}
.opened .line2 {
  stroke-dasharray: 1 60;
  stroke-dashoffset: -30;
  stroke-width: 6;
}
.opened .line3 {
  stroke-dasharray: 90 207;
  stroke-dashoffset: -134;
  stroke-width: 6;
}


@include for-max-1443px() {
  .header {
    &__container {
      gap: 0%;
    }
    .nav {
      width: 53%;
    }
  }
}

@include for-max-1050px() {
  .header {
    .nav {
      height: 10rem;
        display: grid;
        justify-content: center;
        align-items: center;
        justify-content: space-evenly;
        gap: 0;
        grid-template-areas:
        'A A B B'
        'C C D D'
        'E E E E';
      a {
        justify-self: center;
        font-size: 1.2rem;
      }
      .nav-item.a {
        grid-area: A;
        justify-self: center;
      }
      .nav-item.b {
        grid-area: B;
        justify-self: center;
      }
      .nav-item.c {
        grid-area: C;
        justify-self: center;
      }
      .nav-item.d {
        grid-area: D;
        justify-self: center;
        padding: 0;
      }
      .nav-item.e {
        grid-area: E;
        justify-self: center;
        padding: 0;
      }
    }
  }
}

@include for-max-1000px() {
  .header {
    position: relative;
    height: 150px;
    .button {
      width: 250px;
    }
    .nav {
      width: 100%;
      transition: .3s;
      transform: translateX(110%);
      padding: 120px 0px 0px 20px;
      right: 0;
      position: fixed;
      background: lightskyblue;
      display: flex;
      flex-direction: column;
      gap: 15px;
      height: 100vh;
      z-index: 9999;
      align-items: flex-start;
      justify-content: flex-start;
      top: 0;
      box-sizing: border-box;
      a {
        font-size: 2rem;
      }
      &.activated {
        transform: translateX(0%);
        animation: taadaa .3s forwards;
      }
    }
    &__signUp {
      display: none;
      width: calc(100%/3);
      padding: 0;
      transition: .3s;
      &.activated {
        display: flex;
        flex-direction: column;
        position: fixed;
        z-index: 10000;
        left: 20px;
        top: 420px;
        animation: taadaa 1.2s forwards;
        animation-timing-function: ease;
      }
      a {
        width: 50%;
      }
    }
    &__hamburger {
      display: block !important;
      &.opened {
        position: fixed;
        top: 42px;
      }
    }
    .logo {
      width: calc(100%/3);
      .logo1 {
        transition: .3s;
        &.activated {
          z-index: 10000;
          position: fixed;
          animation: taadaa-logo 1.2s forwards;
          animation-timing-function: ease;
        }
        
      }
    }
    
  }
}

@include for-max-740px() {
  .header {
    .logo {
      width: 60%;
      justify-content: flex-start;
      .logo1 {
        transition: .3s;
        &.activated {
          z-index: 10000;
          position: fixed;
          animation: taadaa-logo 1.2s forwards;
          animation-timing-function: ease;
        }
        
      }
    }
    
  }
}

@include for-max-440px() {
  .header {
    &__hamburger {
      right: 2rem;
    }
    .logo {
      width: 70%;
      .logo1 {
        &.activated {
          animation: taadaa-logo1 1.2s forwards;
          animation-timing-function: ease;
        }
      }
    }
  }
 
}

@include for-max-400px() {
  .header {
    &__hamburger {
      right: 1rem;
    }
    .logo {
      width: 90%;
    }
  }
}