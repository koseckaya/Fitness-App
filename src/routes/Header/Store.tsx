import React, { Component } from 'react'
import { Container } from '../../components/Container'

export class Store extends Component {
  render() {
    return (
      <Container>
        Store
      </Container>
    )
  }
}