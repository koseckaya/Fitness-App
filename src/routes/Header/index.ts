export { default as Header } from './Header';
export type { Props as HeaderProps } from './Header';
