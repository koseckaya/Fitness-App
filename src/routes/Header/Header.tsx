
import { FC, useContext } from 'react';
import { Link } from 'react-router-dom';
import { Button } from '../../components/Button';
import { Container } from '../../components/Container';
import { Logo } from '../../components/Logo';
import { UserContext } from '../../components/utils/contexts';
import { DropdownMenu } from '../../components/DropdownMenu';
import './Header.scss'

export type Props = {
    className?: string;
    testId?: string;
};

const Header: FC<Props> = () => {
  const { currentUser } = useContext(UserContext);

  return (
    <div className="header" >
        <Container>
            <div className="header__container" >
                <nav className='nav'>
                    <Link to={`/programs`} className="nav-item">Workout Programs</Link>
                    <Link to={`/videos`} className="nav-item">Workout Videos</Link>
                    <Link to={`/recipes/alcohol-free`} className="nav-item">Recipes</Link>
                    <Link to={`/calculator`} className="nav-item">Calculator</Link>
                    <Link to={`/about`} className="nav-item">About</Link>
                </nav>
                <Link to={`/`} className="logo"> <Logo /> </Link>
                {currentUser ? (
                  <DropdownMenu userName={currentUser}/>
                  /* <Button content={`Sign Out ${name}`} type={Button.TYPES.DEFAULT} 
                    onClick={signOutUser}/> */
                  ) : (
                    <div className='header__signUp'>
                      <Link to={`/signup`}><Button content="Sign Up" type={Button.TYPES.DEFAULT} /></Link>
                      <Link to={`/login`}><Button content="Log In" type={Button.TYPES.PRIMARY} /></Link>
                    </div>
                  )
                }
            </div>
        </Container>
    </div>
  )
};

export default Header;
