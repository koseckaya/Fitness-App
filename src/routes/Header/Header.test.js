import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import Header from './Header';


it('renders correctly', () => {
    const tree = renderer
        .create(<Router><Header /></Router>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});