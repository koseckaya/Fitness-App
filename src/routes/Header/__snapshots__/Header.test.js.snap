// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="header"
>
  <div
    className="container "
  >
    <div
      className="header__container"
    >
      <nav
        className="nav"
      >
        <a
          className="nav-item"
          href="/programs"
          onClick={[Function]}
        >
          Workout Programs
        </a>
        <a
          className="nav-item"
          href="/videos"
          onClick={[Function]}
        >
          Workout Videos
        </a>
        <a
          className="nav-item"
          href="/recipes/alcohol-free"
          onClick={[Function]}
        >
          Recipes
        </a>
        <a
          className="nav-item"
          href="/calculator"
          onClick={[Function]}
        >
          Recipe Analyzer
        </a>
        <a
          className="nav-item"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </nav>
      <a
        className="logo"
        href="/"
        onClick={[Function]}
      >
         
        <div
          className="logo1"
        />
         
      </a>
      <div
        className="header__signUp"
      >
        <a
          href="/signup"
          onClick={[Function]}
        >
          <div
            className="button  "
          >
            Sign Up
          </div>
        </a>
        <a
          href="/login"
          onClick={[Function]}
        >
          <div
            className="button  button-primary"
          >
            Log In
          </div>
        </a>
      </div>
      <div
        className="toggle-container"
      >
        <label
          className="switch"
          htmlFor="toggler"
        >
          <input
            checked={false}
            className="toggle-input"
            id="toggler"
            onClick={[Function]}
            readOnly={true}
            type="checkbox"
          />
          <span
            className="toggle-slider"
          />
          <span
            className="toggle-wave"
          />
        </label>
      </div>
    </div>
  </div>
</div>
`;
